Description: updating makefile
 updating makefile to make it more compatible with packaging
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2020-04-26
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -1,23 +1,33 @@
-CFLAGS = -std=c99 -pedantic
-SDL_FLAGS = `sdl-config --cflags`
-SDL_LIBS = `sdl-config --libs`
-LDLIBS = -lm
-
-TARGETS = Unnamed
-
-all: $(TARGETS) 
-
-Unnamed: Unnamed.o
-	$(CC) $(LDFLAGS) $^ $(LOADLIBES) $(LDLIBS) $(SDL_LIBS) -o $@
-
-Unnamed.o: Unnamed.c
-	@cat ./s.h
-	@echo "`cat $< | perl -pe 's/\s//g' | wc -c` <= 4096"
-	@echo "`ls -l $< | awk '{print $$5}'` <= 8192"
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) $(SDL_FLAGS) -o $@ $<
-
-distclean: clean
-	$(RM) $(TARGETS)
+#variables de compilation
+SHELL = /bin/sh
+CC    := gcc
+PKG_CONFIG:=pkg-config
+CFLAGS   += -std=c99 -Wall -pipe $(shell $(PKG_CONFIG) --cflags sdl)
+LDFLAGS  += $(shell $(PKG_CONFIG) --libs sdl) -lm
+TARGET   = unnamed-by-pirro
+#variable de nettoyage
+RM_F = rm -f
+#variables d'instalation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -m  755
+prefix          = /usr
+exec_prefix     = $(prefix)
+bindir          = $(prefix)/bin
+
+SOURCE          = Unnamed.c
+
+export
+
+all: $(TARGET)
+
+$(TARGET):
+	 $(CC) $(SOURCE) $(CFLAGS) -o $(TARGET) $(LDFLAGS)
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	-@$(RM_F) $(DESTDIR)$(bindir)/$(TARGET)
+	$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(bindir)
 
 clean:
-	$(RM) *.o *~ $(TARGETS)
+	-$(RM_F) $(TARGET)
--- /dev/null
+++ b/s.h
@@ -0,0 +1,4 @@
+#include <SDL/SDL.h>
+#include <math.h>
+#include <stdlib.h>
+
